### PR TITLE
Fill gaps in the readme / docs and fix installation error

### PR DESCRIPTION
**Pull request by connecteev**
This PR fills in a lot of the gaps that I found when I tried to get this working. After hours of trial and error, this setup finally works...I hope it helps others get up-to-speed a lot faster!

1. install npm-sass and node-sass to fix the npm installation error reported here: https://github.com/fritsvt/laravel-nuxt-authentication/issues/6
2. updates .env.example config with missing values
3. Adds detailed and fool-proof setup instructions to the readme
4. Add buttons and config to log in with twitter and linkedin
5. Add fix for Incorrect / dummy email saved to DB after social login with github or linkedin, issue reported here: https://github.com/fritsvt/laravel-nuxt-authentication/issues/8

### CODE REVIEW

File: client/components/SocialLogin.vue [hunk 1]
```diff
@@ -9,6 +9,12 @@
         <button @click="socialLogin('facebook')" class="social-login-btn facebook">
             <i class="fab fa-facebook"></i>
         </button>
+        <button @click="socialLogin('twitter')" class="social-login-btn twitter">
```
Comment (fritsvt) [inline client/components/SocialLogin.vue]: Why did you change these to twitter and linkedin? That doesn't match the Youtube content. (I uploaded this repo because people asked for it because of a youtube tutorial series I did)

File: .env.example [hunk 2]
```diff
@@ -38,14 +39,29 @@ PUSHER_APP_CLUSTER=mt1
...
 MIX_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"
 
-GITHUB_CLIENT_ID=
-GITHUB_CLIENT_SECRET=
-GITHUB_REDIRECT_URL=
 
-GOOGLE_CLIENT_ID=
-GOOGLE_CLIENT_SECRET=
-GOOGLE_REDIRECT_URL=
+# Set these up at https://github.com/settings/applications/new
```
Comment (fritsvt) [inline .env.example]: Like this a lot! Good idea to add these urls to the env

File: app/Http/Controllers/api/Auth/SocialLoginController.php [hunk 3]
```diff
@@ -28,15 +28,11 @@ public function callback($service)
     {
         try {
             $serviceUser = Socialite::driver($service)->stateless()->user();
+            $email = $serviceUser->getEmail();
```
Comment (fritsvt) [inline app/Http/Controllers/api/Auth/SocialLoginController.php]: What was the reason for moving the email call here? I don't think the getEmail function would ever throw an exception.

File: client/package.json [hunk 4]
```diff
@@ -16,7 +16,9 @@
     "bootstrap": "^4.1.3",
     "bootstrap-vue": "^2.0.0-beta",
     "cross-env": "^5.2.0",
-    "nuxt": "^2.0.0"
+    "npm-sass": "^2.3.0",
```
Comment (fritsvt) [inline client/package.json]: Why did you have to add npm-sass and sass-loader? I don't think I needed those but please let me know why those we're required in your case.

File: readme.md [hunk 5]
```diff
@@ -1,36 +1,154 @@
-# Local and social authentication with Laravel and Nuxt
```
Comment (fritsvt) [inline readme.md]: Thanks a lot for this, looks really good and it will hopefully save people a lot of time 👍

**Comment by fritsvt** [comment]
Thanks a lot for your PR. I have some questions and possible changes before merging to master. I'm really happy about the new instructions and some other changes you made so thank you.

File: .env.example (hunk 2 above)
Comment (connecteev) [reply .env.example]: Took me a while to get this working, so I'm glad you like it @fritsvt

File: app/Http/Controllers/api/Auth/SocialLoginController.php (hunk 3 above)
Comment (connecteev) [reply app/Http/Controllers/api/Auth/SocialLoginController.php]: ah, ok...I can take it out of the try block. I thought it could fail / throw an exception but I havent dug into the socialite docs.

File: client/package.json (hunk 4 above)
Comment (connecteev) [reply client/package.json]: @fritsvt This is needed to fix this
https://github.com/fritsvt/laravel-nuxt-authentication/issues/6

File: readme.md (hunk 5 above)
Comment (connecteev) [reply readme.md]: @fritsvt my pleasure 👍 👍

File: app/Http/Controllers/api/Auth/SocialLoginController.php [hunk 6]
```diff
@@ -66,13 +62,8 @@ public function needsToCreateSocial(User $user, $service)
 
     public function getExistingUser($serviceUser, $email, $service)
     {
-        if ($service == 'google') {
```
Comment (fritsvt) [inline app/Http/Controllers/api/Auth/SocialLoginController.php]: Please see my comment on #8

File: client/components/SocialLogin.vue (hunk 1 above)
Comment (connecteev) [reply client/components/SocialLogin.vue]: @fritsvt that's a fair point. I am trying to get all the major social logins working:
google, linkedin, facebook, twitter being the main ones. I understand if you don't want to add them though I don't think people will complain with the added functionality. Your call.

File: app/Http/Controllers/api/Auth/SocialLoginController.php (hunk 6 above)
Comment (connecteev) [reply app/Http/Controllers/api/Auth/SocialLoginController.php]: @fritsvt I see...just replied to your comment, let me know your thoughts

File: client/components/SocialLogin.vue (hunk 1 above)
Comment (fritsvt) [reply client/components/SocialLogin.vue]: It would be ideal if we somehow dynamically showed them when the services are active. I didn't realize you kept the others in though but Ill gladly accept these two new services. Will hopefully show people how easy it should be to add new services.

Comment (connecteev) [reply client/components/SocialLogin.vue]: I think that's a good call @fritsvt

File: app/Http/Controllers/api/Auth/SocialLoginController.php (hunk 3 above)
Comment (connecteev) [reply app/Http/Controllers/api/Auth/SocialLoginController.php]: @fritsvt done

**Comment by connecteev** [comment]
done, I moved getEmail() call out since it doesnt throw an exception

**Comment by fritsvt** [comment]
> done, I moved getEmail() call out since it doesnt throw an exception

Cool, thanks. I'd still like to solve #8 first before merging. We can find a new solution, move back to the old way for now or I can see if I find some time soon and make something. Of course you're also free to help with a new solution if you're still up to work on this.

**Comment by connecteev** [comment]
> > done, I moved getEmail() call out since it doesnt throw an exception
> 
> Cool, thanks. I'd still like to solve #8 first before merging. We can find a new solution, move back to the old way for now or I can see if I find some time soon and make something. Of course you're also free to help with a new solution if you're still up to work on this.

Just replied to #8 -- I prefer keeping it as is, and proposed a workaround....but it's your repo, so your call.

**Comment by fritsvt** [comment]
Thanks for the reply, I'd prefer to remove it for now until we have a solution. I'd rather have someone discover why it's .local then them deploying it and finding out accounts can be hijacked by others.

**Comment by connecteev** [comment]
Sure thing...and what you said sounds fair. Do you want to make that change to the PR and merge this in?
Any plans to maintain and enhance the repo, btw? Would like to improve this further and have ideas if you do.

**Comment by fritsvt** [comment]
Yeah, I'm not sure how much time I have to work on it but I'd like to fix this issue at least. Would love to hear about your ideas though

**Comment by connecteev** [comment]
Ok, I'll change the PR back - may add a config setting, defaulting to "dont pull the email if unverified by the provider" - that way my branch stays in sync with yours.

As for ideas, we're 80% of the way there with this repo. Like you, I have a lot going on so without spending a lot of time right now describing in great detail, it could involve 
1. authenticated API tokens/requests from the nuxt client, basically preventing unauthorized use of the API after login. For example: after a user logs in, the nuxt client would request data over Axios/Ajax for a logged-in user than no one else should be able to access
2. during social login, option to pull data from expanded scopes from each source
3. for plain auth: account confirmation within X days or account gets deactivated (see devdojo.com on how they do this, it's slick), basic password reset / recovery, etc
Scratching the surface here and if any of the above is interesting we can explore - otherwise no harm no foul.

**Comment by connecteev** [comment]
done
